docs/enhancements/api-for-skipping-validations.md

```diff
@@ -0,0 +1,139 @@
+This api should have it's own feature flag to allow us to disable it.
+
+It should be possible to use this API in an unsupported way, it is currently undecided whether or not this will be open or not.
+An event should be emitted whenever the API is used during an installation, this can be used to drive signatures etc.
+# Motivation
+
+Some customer and partners have very specific HW that doesn't pass our validations and we want to allow them to proceed with installation.
+Sometimes we have bugs in our validations that block people from installing and we don't want our partners left in a position where they can't install.
+
+### Goals
+
+- Allow validations to be bypassed so that installation may proceed.
+
+### Non-Goals
+
+- We should not add uneccesary noise to triage tickets with this, it is therefore quite important to add some signatures to indicate that these validations have been bypassed.
+
+## Proposal
+
+I think the API should be kept reasonably simple.
+
+### List existing validations.
+
+GET api/v2/skippedValidations
+Will return a JSON array of all the validations that are currently being skipped, format similar to
+
+```
+{
+  "skipped-host-validations": [
+    "no-ip-collisions-in-network",
+    "sufficient-network-latency-requirement-for-role'"
+  ],
+  "skipped-cluster-validations": [
+    "network-prefix-valid",
+    "api-vips-defined"
+  ]
+}
+```
+
+### Add some skipped validations.
+
+POST api/v2/skippedValidations
+
+Essentially a document will be submitted via POST in JSON
+```
+{
+  "skipped-host-validations": [
+    "no-ip-collisions-in-network"
+  ],
+  "skipped-cluster-validations": [
+    "api-vips-defined"
+  ]
+}
+```
+
+* Any entries found will be added to the existing list of validations to be skipped. This will appropriately de-deuplicated and no error will be returned if duplication is attempted.
+* If an attempt is made to reference a validation that does not exist then a warning will be logged but this will not be considered fatal.
+* After the call has completed, if this leads to a change in the collection of skipped validations then an event will be emitted.
+
+### Delete some skipped validations.
+
+DELETE api/v2/skippedValidations
+
+Essentially a document will be submitted via POST in JSON
+```
+{
+  "skipped-host-validations": [
+    "no-ip-collisions-in-network"
+  ],
+  "skipped-cluster-validations": [
+    "api-vips-defined"
+  ]
+}
+```
+* Any entries found will be deleted from the existing list of validations to be skipped.
+* If an attempt is made to reference a validation that does not exist then a warning will be logged but this will not be considered fatal.
+* If a validation has not been skipped already and is marked for removal then this entry will be ignored and not emit an error.
+* After the call has completed, if this leads to a change in the collection of skipped validations then an event will be emitted.
+
+### Delete all skipped validations
+
+DELETE api/v2/skippedValidations/all
```

Comment:
DELETE ``.../all`` is not RESTful, maybe instead deleting the whole ``skippedValidations`` resource and treating it with ``PATCH/PUT`` requests instead of the ``POST`` requests

Reply:
Accpted point about the delete operation.

Worth considering a bit of both.

If we reserve the  POST request for the _addition_ of another "skipped validation" and keep the patch for clearing or bulk adding validations, this might be best. 

The advantage of having the additional POST operation as described is that it is very easy to skip "one more of.." the validations without having to edit JSON etc.